### ISSUE TITLE
upgrade nimbusds dependency with spring security oauth2 client 5.1.x

### ISSUE THREAD

**Issue by Incarnation-p-lee** [AzureAD/azure-activedirectory-library-for-java#248]
when leverage spring security 5.1.x with this package, there may be dependency conflict for nimbusds. This package leverage `5.64.4`, and spring security leverage 6.0. By the rules of maven dependency, it may chose 6.0 version based on you tree of dependencies.
```pom
		<dependency>
			<groupId>com.nimbusds</groupId>
			<artifactId>oauth2-oidc-sdk</artifactId>
			<version>5.64.4</version>
		</dependency>
```

Unfortunately, if maven chose 6.0 for you, the source code of `AdalTokenRequest` may met CastException. The 6.0 version changes the API parameter to `Map<String, List<String>` instead of 
`Map<String, String>`.
```
    AdalOAuthRequest toOAuthRequest() throws SerializeException {

        if (this.uri == null) {
            throw new SerializeException("The endpoint URI is not specified");
        }

        final AdalOAuthRequest httpRequest = new AdalOAuthRequest(
                HTTPRequest.Method.POST, this.uri, headerMap, this.proxy,
                this.sslSocketFactory);
        httpRequest.setContentType(CommonContentTypes.APPLICATION_URLENCODED);
        final Map<String, String> params = this.authzGrant.toParameters();
        httpRequest.setQuery(URLUtils.serializeParameters(params)); // Cast Exception in serialize.
        if (this.clientAuth != null) {
            this.clientAuth.applyTo(httpRequest);
        }

        return httpRequest;
    }
```
As stated above, is there any plan to upgrade the nimbusds to 6.0 for support spring security 5.1.x ?

**Comment by mkdirJava** [comment]
@Incarnation-p-lee 
I feel like I am missing something 
httpRequest.setQuery(URLUtils.serializeParameters(params)); // Cast Exception in serialize.

comes from the lib;
    com.microsoft.aad.adal4j
in turn uses this lib
    oauth2-oidc-sdk ( I believe version 6 is used in the build as it is named "oauth2-oidc-sdk-6.0.jar")

the exception originates in oauth2-oidc-sdk as 
          public static String serializeParameters(Map<String, List<String>> params)
and it has line 101 ( Where the error happens) as

                         for (String value : (List)entry.getValue())

https://bitbucket.org/connect2id/oauth-2.0-sdk-with-openid-connect-extensions/src/35e3a898d4fd29d406cfb46fe579784a209c1ae2/src/main/java/com/nimbusds/oauth2/sdk/util/URLUtils.java?at=5.64.4&fileviewer=file-view-default

The value of the map is a String and not a list as stated in the exception however going up versions of 
oauth2-oidc-sdk will only still have the value of the map as a List and not as a String. 

Any pointers would be grateful 
@psignoret 
@sandermak

**Comment by mkdirJava** [comment]
@Incarnation-p-lee 
Your comment on previous 

When using Spring Boot 2.1.0 with azure-active-directory-spring-boot-starter get error on login page Microsoft/azure-spring-boot#457

Sounds more interesting with a missing constructor?

java.lang.NoSuchMethodError: org.springframework.security.oauth2.core.oidc.user.DefaultOidcUser.<init>(Ljava/util/Set;Lorg/springframework/security/oauth2/core/oidc/OidcIdToken;Ljava/lang/String;)V

I may be barking up the wrong tree here.

**Comment by lpfeup** [comment]
Any updates regarding this issue?

I am having the same version conflicts, but with [pac4j-oidc](https://github.com/pac4j/pac4j/tree/master/pac4j-oidc), which depends on `oauth2-oidc-sdk` `v6.5`.

Had to pin `oauth2-oidc-sdk` to `v5.64.4` on `pom.xml` as temporary workaround.

**Comment by sangonzal** [comment]
Apologies for the delay - we have create an item on the backlog to update `oauth2-oidic-sdk` version `6.5`. These changes should be completed by the end of the week. I will update this issue once a new release is out.

**Comment by sangonzal** [comment]
This has been included as part of  v1.6.4
